htdocs/web_portal/javascript/leafletembed.js

```diff
@@ -1,46 +1,37 @@
-//function to display google map. Adapted by George Ryall from the following tutorials:
-// * https://developers.google.com/maps/articles/phpsqlajax_v3
-// * https://developers.google.com/maps/documentation/javascript/tutorial
-// * http://www.w3schools.com/googleAPI/
-// * The example code here : http://gmaps-samples-v3.googlecode.com/svn/trunk/xmlparsing/ 
+// javascript file to display a leaflet map.
+// Based on https://leafletjs.com/examples/quick-start/ and
+// https://switch2osm.org/using-tiles/getting-started-with-leaflet/
 
-google.maps.event.addDomListener(window, 'load', initialize);
-
-var infowindow;
 var map;
 
-function initialize() {
-    var mapProp = {
-        //Starting position of map
-        center:new google.maps.LatLng(30,0),
-        zoom:2,
-        //Enable zoom control, but move it to the left and make it small
-        zoomControl:true,
-        zoomControlOptions: {
-            style:google.maps.ZoomControlStyle.SMALL,
-            position:google.maps.ControlPosition.RIGHT_TOP
-        },
-        //Allow map type choiuce, but move to bottom left
-        mapTypeControl:true,
-        mapTypeControlOptions: {
-            position:google.maps.ControlPosition.TOP_RIGHT   
-        },
-        //Turn off pan control - it clutters map and users can drag if needed
-        panControl:false,
-        //turn of street view controller
-        streetViewControl:false,
-        //Options:ROADMAP/SATELLITE/HYBRID/TERRAIN   
-        mapTypeId:google.maps.MapTypeId.ROADMAP
-    };
+// Function to display a leaflet map with markers from GOCDB Site data.
+function initmap() {
+	// set up the map
+	map = new L.Map('map');
+
+	// create the tile layer with correct attribution.
+	var osmUrl='https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png';
+	var osmAttrib='Map data © <a href="https://openstreetmap.org">OpenStreetMap</a> contributors';
+	var osm = new L.TileLayer(osmUrl, {
+        attribution: osmAttrib,
+        // A min zoom of 2 gives you a view of most of the world.
+        minZoom: 2,
+        // A max zoom of more than 20 doesn't have any map tiles.
+        maxZoom: 19,
+    });		
 
-    map = new google.maps.Map(document.getElementById("GoogleMap"), mapProp);
+	// Centre the view on RAL, at zoom level 2.
+	map.setView(new L.LatLng(51.5738, -1.3146), 2);
```

Comment:
Tempting, but it might be best to use something similar to the previous centre as that seemed to keep all the sites in view.

Reply:
Best centre ever.

Reply:
I have re-centred to something similar to what we have now - most (if not all) sites are now visible in the default view